File: web/lib/flowers.ts
// Flower-type catalog — MIRRORS backend/app/core/listings/flowers.py (the authoritative
// fixed list). Stable ascii id ↔ RU label («Передарим» copy). The form/filter shows
// labels but submits/filters by id; the listing card/detail map ids → labels for display.
// Order is meaningful (form/filter display order, API_CONTRACT §1).

export const FLOWERS: readonly { id: string; label: string }[] = [
  { id: 'roses', label: 'Розы' },
  { id: 'peony_roses', label: 'Пионовидные розы' },
  { id: 'peonies', label: 'Пионы' },
  { id: 'tulips', label: 'Тюльпаны' },
  { id: 'hydrangea', label: 'Гортензия' },
  { id: 'chrysanthemums', label: 'Хризантемы' },
  { id: 'eustoma', label: 'Эустома' },
  { id: 'ranunculus', label: 'Ранункулюсы' },
  { id: 'alstroemeria', label: 'Альстромерия' },
  { id: 'lilies', label: 'Лилии' },
  { id: 'wildflowers', label: 'Полевые' },
];

const LABEL = new Map(FLOWERS.map((f) => [f.id, f.label]));

/** id → RU label; falls back to the raw id if unknown (forward-compatible). */
export function flowerLabel(id: string): string {
  return LABEL.get(id) ?? id;
}

/** Map a listing's flower_types (ids) to display labels, dropping unknowns. */
export function flowerLabels(ids: string[]): string[] {
  return ids.map(flowerLabel);
}

File: backend/app/core/listings/flowers.py
"""Flower-type reference — a fixed, ordered list (export-next.md §2).

Canonical enum for the «тип цветов» field on a listing and the catalog filter.
Order is meaningful (it is the form/filter display order). The IDs are stable
ascii slugs; the RU labels are customer-facing copy («Передарим»).
"""

from __future__ import annotations

# (id, RU label) — order preserved; this list IS the source of truth.
FLOWERS: tuple[tuple[str, str], ...] = (
    ("roses", "Розы"),
    ("peony_roses", "Пионовидные розы"),
    ("peonies", "Пионы"),
    ("tulips", "Тюльпаны"),
    ("hydrangea", "Гортензия"),
    ("chrysanthemums", "Хризантемы"),
    ("eustoma", "Эустома"),
    ("ranunculus", "Ранункулюсы"),
    ("alstroemeria", "Альстромерия"),
    ("lilies", "Лилии"),
    ("wildflowers", "Полевые"),
)

FLOWER_IDS: frozenset[str] = frozenset(fid for fid, _ in FLOWERS)

_LABELS: dict[str, str] = dict(FLOWERS)


def label(flower_id: str) -> str | None:
    return _LABELS.get(flower_id)
